### PR TITLE
feat(rest): endpoint to get attachmentUsages for a project.

### LINKED ISSUE #2094: Attachment Usage endpoint for project page
### Description:

New endpoint to get AttachmentUsage Information for a project.

### PR DESCRIPTION
**Pull request by rudra-superrr**
Issue: closes #2094 

Description: updated the existing attachmentUsages endpoint with more fields in response structure.

### CODE REVIEW

**Comment by smrutis1** [comment]
Code looks ok. But there are some additional fields in the response which is not required for the page. I believe.
image

It would be good if that can be removed.

Please refer this to ignore the booleans in the response.
https://github.com/eclipse-sw360/sw360/blob/b58f28c317fad35b4d3e35f17a1d8a107adcca7c/rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/JacksonCustomizations.java#L377

**Comment by rudra-superrr** [comment]
Hi @smrutis1, additional fields removed.

**Review by smrutis1 — CHANGES_REQUESTED** [review]
Requesting some minor changes

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java [hunk 1]
```diff
@@ -1392,12 +1406,42 @@ public ResponseEntity<CollectionModel<EntityModel<Project>>> getUsedByProjectDet
...
+        ObjectMapper oMapper = new ObjectMapper();
+        Map<String, ProjectReleaseRelationship> releaseIdToUsages = sw360Project.getReleaseIdToUsage();
+        Map<String, Object> projectMap = oMapper.convertValue(sw360Project, Map.class);
+        Map<String, Object> resultMap = new HashMap<>();
+        resultMap.put("linkedProjects", (Map<String, Object>) projectMap.get("linkedProjects"));
+        resultMap.put("releaseIdToUsage", (Map<String, Object>) projectMap.get("releaseIdToUsage"));
+
+        Map<String, Object> releaseIdToUsage = (Map<String, Object>) resultMap.get("releaseIdToUsage");
+        Set<String> fieldsToRemove = new HashSet<>();
+        fieldsToRemove.add("setCreatedBy");
```
Comment (smrutis1) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:1424]: I was expecting something like this at class level.

`public static final ImmutableSet<String> SET_OF_FIELDS_TO_REMOVE = ImmutableSet.of("setCreatedBy", "setCreatedOn", "setComment", "setReleaseRelation", "setMainlineState");`

Comment (rudra-superrr) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java]: Understood, made the changes.

Comment (smrutis1) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java]: Thanks for the changes, I was expecting it at class level, but it should be fine.

**Review by smrutis1 — APPROVED** [review]
Code looks good. Tested successfully.

**Pull request merged**